airbyte-webapp/src/components/connection/CatalogTree/next/StreamDetailsPanel/StreamFieldsTable/StreamFieldsTable.tsx

```diff
@@ -28,32 +36,48 @@ export interface TableStream {
 
 export interface StreamFieldsTableProps {
   config?: AirbyteStreamConfiguration;
+  handleFieldToggle: (fieldPath: string[], isSelected: boolean) => void;
   onCursorSelect: (cursorPath: string[]) => void;
   onPkSelect: (pkPath: string[]) => void;
   shouldDefinePk: boolean;
   shouldDefineCursor: boolean;
   isCursorDefinitionSupported: boolean;
   isPKDefinitionSupported: boolean;
   syncSchemaFields: SyncSchemaField[];
+  toggleAllFieldsSelected: () => void;
 }
 
 export const StreamFieldsTable: React.FC<StreamFieldsTableProps> = ({
   config,
+  handleFieldToggle,
   onPkSelect,
   onCursorSelect,
   shouldDefineCursor,
   shouldDefinePk,
   isCursorDefinitionSupported,
   isPKDefinitionSupported,
   syncSchemaFields,
+  toggleAllFieldsSelected,
 }) => {
   const { formatMessage } = useIntl();
-
-  const isCursor = useMemo(() => (path: string[]) => equal(config?.cursorField, path), [config?.cursorField]);
-  const isPrimaryKey = useMemo(
-    () => (path: string[]) => !!config?.primaryKey?.some((p) => equal(p, path)),
+  const isColumnSelectionEnabled = useExperiment("connection.columnSelection", false);
+  const isCursor = useCallback((path: string[]) => equal(config?.cursorField, path), [config?.cursorField]);
+  const isPrimaryKey = useCallback(
+    (path: string[]) => !!config?.primaryKey?.some((p) => equal(p, path)),
     [config?.primaryKey]
   );
+  const checkIsFieldSelected = useCallback(
+    (field: SyncSchemaField): boolean => {
+      // All fields are implicitly selected if field selection is disabled
+      if (!config?.fieldSelectionEnabled) {
+        return true;
+      }
+
+      // path[0] is the top-level field name for all nested fields
+      return !!config?.selectedFields?.find((f) => isEqual(f.fieldPath, [field.path[0]]));
```

Comment:
Nit: You could also use `.some` instead of `!!object.find`

Reply:
True, I will leave this though since I don't think it makes any functional difference